packages/flutter/lib/src/material/autocomplete.dart

```diff
@@ -0,0 +1,256 @@
+/// }
+///
+/// class AutocompleteBasicUserExample extends StatelessWidget {
+///   AutocompleteBasicUserExample({Key? key}) : super(key: key);
+///
+///   static final List<User> _userOptions = <User>[
+///     User(name: 'Alice', email: 'alice@example.com'),
+///     User(name: 'Bob', email: 'bob@example.com'),
+///     User(name: 'Charlie', email: 'charlie123@gmail.com'),
+///   ];
+///
+///   static String _displayStringForOption(User option) => option.name;
+///
+///   @override
+///   Widget build(BuildContext context) {
+///     return Autocomplete<User>(
+///       displayStringForOption: _displayStringForOption,
+///       optionsBuilder: (TextEditingValue textEditingValue) {
+///         if (textEditingValue.text == '') {
+///           return const Iterable<User>.empty();
+///         }
+///         return _userOptions.where((User option) {
+///           return option.toString().contains(textEditingValue.text.toLowerCase());
+///         });
+///       },
+///       onSelected: (User selection) {
+///         print('You just selected ${_displayStringForOption(selection)}');
+///       },
+///     );
+///   }
+/// }
+/// ```
+/// {@end-tool}
+class Autocomplete<T extends Object> extends StatefulWidget {
+  /// Creates an instance of [Autocomplete].
+  const Autocomplete({
+    Key? key,
+    required this.optionsBuilder,
+    this.displayStringForOption = RawAutocomplete.defaultStringForOption,
+    this.fieldViewBuilder = _defaultFieldViewBuilder,
+    this.onSelected,
+    this.optionsViewBuilder,
+  }) : assert(displayStringForOption != null),
+       assert(optionsBuilder != null),
+       super(key: key);
+
+  /// {@macro flutter.widgets.RawAutocomplete.displayStringForOption}
+  final AutocompleteOptionToString<T> displayStringForOption;
+
+  /// {@macro flutter.widgets.RawAutocomplete.fieldViewBuilder}
+  ///
+  /// If not provided, will build a standard Material-style text field by
+  /// default.
+  final AutocompleteFieldViewBuilder fieldViewBuilder;
+
+  /// {@macro flutter.widgets.RawAutocomplete.onSelected}
+  final AutocompleteOnSelected<T>? onSelected;
+
+  /// {@macro flutter.widgets.RawAutocomplete.optionsBuilder}
+  final AutocompleteOptionsBuilder<T> optionsBuilder;
+
+  /// {@macro flutter.widgets.RawAutocomplete.optionsViewBuilder}
+  ///
+  /// If not provided, will build a standard Material-style list of results by
+  /// default.
+  final AutocompleteOptionsViewBuilder<T>? optionsViewBuilder;
+
+  static Widget _defaultFieldViewBuilder(BuildContext context, TextEditingController textEditingController, FocusNode focusNode, VoidCallback onFieldSubmitted) {
+    return _AutocompleteField(
+      focusNode: focusNode,
+      textEditingController: textEditingController,
+      onFieldSubmitted: onFieldSubmitted,
+    );
+  }
+
+  @override
+  _AutocompleteState<T> createState() => _AutocompleteState<T>();
+}
+
+class _AutocompleteState<T extends Object> extends State<Autocomplete<T>> {
```

Comment:
Does this have to be stateful?

Reply:
Ah thank you, how did I even miss that...